monymony-public/monymony-eslint-config
# build: support eslint version 9 (flatten config) and setup rollup bundler

## description
To support [ESLint 9](https://eslint.org/blog/2022/08/new-config-system-part-1/), I migrated the configuration settings to support ESLint 9 using the flat config format and bundled the package with Rollup.


## Test Plan
1. Pull this PR, install it locally, and test the package by specifying the path as shown below in your project’s package.json

```package.json
{
   "devDependencies": {
          "@monymony/eslint-config": "file:../monymony-eslint-config" // Path to the local eslint package relative to the test project
    }
}

```

2. build the eslint project in local environment.

```sh
yarn build
```

3. Install  needed dependencies  in your project.
```
yarn add -D eslint@latest eslint-plugin-import@latest eslint-import-resolver-typescript@latest typescript-eslint@latest
```

4. Delete the existing .eslintrc file(or other eslint config file) and create a new eslint.config.mjs file.
	•	If your project is already ESM, create the file as eslint.config.js instead.
	•	To check if the project is ESM, verify if the type field in package.json is set to module.

5. Add the following code to the eslint.config.(m)js file. Update the path to backend based on the project structure if necessary.

```
// Your desired lint module (eg. reactNative, backend)
import reactNative from '@monymony/eslint-config/react-native'

export default [
  ...reactNative,
  {
    ignores: ["**/node_modules/*"], // Files to exclude from ESLint are added here instead of using .eslintignore
  },
] 
```

## Future Action
Once local testing is complete, I plan to optimize the package and revise the export format.
